cpp/velox/substrait/SubstraitToVeloxPlan.cc

```diff
@@ -1338,6 +1342,18 @@ void SubstraitToVeloxPlanConverter::separateFilters(
   for (const auto& scalarFunction : scalarFunctions) {
     auto filterNameSpec = SubstraitParser::findFunctionSpec(functionMap_, scalarFunction.function_reference());
     auto filterName = SubstraitParser::getSubFunctionName(filterNameSpec);
+    // Add all decimal filters to remaining functions because their pushdown are not supported.
+    if (format == dwio::common::FileFormat::ORC && scalarFunction.arguments().size() > 0) {
+      auto value = scalarFunction.arguments().at(0).value();
+      if (value.has_selection()) {
+        uint32_t fieldIndex = SubstraitParser::parseReferenceSegment(value.selection().direct_reference());
+        auto type = value.selection().direct_reference().struct_field().field();
+        if (!veloxTypeList.empty() && veloxTypeList.at(fieldIndex)->isDecimal()) {
```

Comment:
Thanks so much for your fix!
According to the PR description, trying to push down filter in timestamp type can also fail. Shouldn't we also exclude it here?

Reply:
@PHILO-HE  At present, when gluten encounters an orc read with timestamp type in the schema, it will fallback to java scan, so if there is timestamp in the filter, native scan will not be performed. ```Validation failed for class io.glutenproject.backendsapi.velox.BackendSettings$ due to: data type TimestampType. in file schema.```